bool success
int32 nr
RobotData[] fdata